### PR TITLE
test(client): add max headers test case

### PR DESCRIPTION
**Pull request by dnlup**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -456,7 +456,7 @@ class Parser {
     this.upgrade = false
     this.headers = []
     this.headersSize = 0
-    this.headersMaxSize = client[kMaxHeadersSize] = 8 * 1024
+    this.headersMaxSize = client[kMaxHeadersSize] || 8 * 1024
```
Comment (dnlup) [inline lib/client.js:459]: Otherwise it would not be possible to pass a custom size from the `Client`.

**Comment by dnlup** [comment]
All the tests here are failing because of the checks on the `asserts` in the headers hooks. Should we be able to pass a custom size to the client? Maybe I took the wrong approach.

File: test/client-errors.js [hunk 2]
```diff
@@ -994,3 +994,29 @@ test('socket errors', t => {
     t.end()
   })
 })
+
+test('headers overflow', { only: true }, t => {
```
Comment (ronag) [inline test/client-errors.js:998]: only?

**Comment by ronag** [comment]
Landed in https://github.com/nodejs/undici/commit/6181944109728944c3213bc1c8c7c5513e29684a

**Comment by dnlup** [comment]
Thank you 😉

**Pull request closed without merging**